project/__init__.py

```diff
@@ -25,10 +26,19 @@ class BaseProject(object):
         'Gemfile': 'bundle update',
         'package.json': 'npm update',
     }
+    
+    '''Amount of seconds to delay test execution after target_url is available.
+    This is due to the fact that some apps take longer to really deploy than what is reported back to the status check.
+    '''
+    TEST_DELAY = 10
 
     def __init__(self, name):
         self.name = name
         self.builddir = os.path.join(TEMPLATEDIR, self.name, 'build/')
+        self.github_token = os.getenv('GITHUB_TOKEN', None) # Maybe add a validation logic to raise an error if not present ?
```

Comment:
Fail gracefully, please. :smile:  I'm not sure what gracefully is in this case, though.

Is there a way to support it as a command line parameter in addition to an env var?  I hate having env references this deep in the code.  It makes testing and development much harder.

Reply:
For the failing gracefully part, at most we could print to `Cannot execute this action because the token is not set` and fail the action.

A command line parameter could be supported on a per subcommand basis but not on operations that run the whole chain. What I mean is, it can be supported for `test:wordpress` which runs ***only*** the testing portion but not for `full:wordpress` which executes the whole command chain of `cleanup`, `init`, `update`, `platformify`, `branch`, `pull`, `pull_request`, `test`, `merge`.

Reply:
It's fairly standard (again, this wants to run in CI) to supply authentication tokens as environment variables. Not sure where the "much harder" here comes from, as long as we're failing gracefully and explicitly if the variable is not supplied.

Reply:
Having the data come in via an env var is fine; it's having the env var that deep in the code that bothers me.  I would be more comfortable with the env var being mapped to some config value early on (Global in dodo.py? Not sure), and then the task reading from that.  As for failing gracefully, what happens if GITHUB_TOKEN is not set?  If there is a stack trace anywhere in the output, then it's not graceful.  It needs to be checked, caught, and a useful error messages given instead.  

"Die the way None would normally die" is not graceful.